src/kcas.mli

```diff
@@ -239,24 +304,20 @@ module Xt : sig
   val call : 'a tx -> xt:'x t -> 'a
   (** [call ~xt tx] is equivalent to [tx.Xt.tx ~xt]. *)
 
-  val attempt : ?mode:Mode.t -> 'a tx -> 'a
-  (** [attempt tx] attempts to atomically perform the transaction over shared
-      memory locations recorded by calling [tx] with a fresh explicit
-      transaction log.  If used in {!Mode.obstruction_free} may raise
-      {!Mode.Interference}.  Otherwise either raises [Exit] on failure to commit
-      the transaction or returns the result of the transaction.  The default for
-      [attempt] is {!Mode.lock_free}. *)
```

Comment:
Good question!  I usually try to comment on changes like this, but I apparently missed that.

I decided to remove `attempt` for two reasons:

1. `attempt` can not implement the blocking mechanism by itself.  Instead it would potentially raise the `Later` exception for the user to handle and then it would mean that there should probably be some additional support to be able to await for changes to multiple locations.

2. I initially provided `attempt` as a means for users to write their own version of `commit` (e.g. with additions like timeouts). However, one should be able to achieve pretty much everything via `commit` already (including timeouts &mdash; e.g. setup a location that is written at timeout) and I don't see much use cases for `attempt` &mdash; I never used it myself except in tests.

So, rather than add even more (potentially unused) things to the API, I decided that it is better to just remove `attempt`.  Something like it can be later added back if there are real use cases for it.

Reply:
I am interested to see how you add a timeout (this also seems like it could be a good example).